lib/server.js

```diff
@@ -152,39 +152,32 @@ class Server extends KarmaEventEmitter {
         this._injector.invoke(watcher.watch)
       }
 
-      const startWebServer = () => {
-        webServer.listen(config.port, config.listenAddress, () => {
-          this.log.info(`Karma v${constant.VERSION} server started at ${config.protocol}//${config.listenAddress}:${config.port}${config.urlRoot}`)
-
-          this.emit('listening', config.port)
-          if (config.browsers && config.browsers.length) {
-            this._injector.invoke(launcher.launch, launcher).forEach((browserLauncher) => {
-              singleRunDoneBrowsers[browserLauncher.id] = false
-            })
-          }
-          if (this.loadErrors.length > 0) {
-            this.log.error('Found %d load error%s', this.loadErrors.length, this.loadErrors.length === 1 ? '' : 's')
-            process.exitCode = 1
-            process.kill(process.pid, 'SIGINT')
-          }
-        })
-      }
-
-      if (fs.existsSync(karmaJsPath) && fs.existsSync(contextJsPath)) {
-        startWebServer()
-      } else {
-        this.log.info('Front-end scripts not present. Compiling...')
```

Comment:
So this test does not work?
https://github.com/karma-runner/karma/blob/31d9a08892f51308efbe7c467c379b3f70f1e416/test/unit/server.spec.js#L113

Reply:
Now that I look more carefully the `else` case test is not correct because `fs` is not injected and the test runs in the same filesystem location every time.  If the file system has the static files from a previous run or from a use of karma the test will pass even if bundling code does not work.

Reply:
I've updated it a bit, now everything should be tested.